src/main/java/com/redhat/cloud/notifications/GwResource.java

```diff
@@ -237,8 +242,21 @@ private static String buildResponseEntity(boolean success, String details) {
         return response.encode();
     }
 
-    private boolean checkIfOCMEventAndIfOrgIdIsGranted(RestAction ra) {
-        return ra.bundle.equals("openshift") && ra.application.equals("cluster-manager") &&
-            restrictAccessByOrgId && !allowedOrgIdList.contains(ra.getOrgId());
+    private boolean checkIfOCMEventAndIfOrgIdIsGranted(RestAction ra, RhIdPrincipal rhIdPrincipal) {
+        if (ra.bundle.equals("openshift") && ra.application.equals("cluster-manager")) {
+            if ("X509".equals(rhIdPrincipal.getType())) {
+                try {
+                    GatewayCertificate gatewayCertificate = restValidationClient.validateCertificateAccordingBundleAndApp(ra.getBundle(), ra.getApplication(), rhIdPrincipal.getName());
+                    Log.infof("Certificate validated, coming from %s", gatewayCertificate.environment);
+                } catch (Exception ex) {
```

Comment:
I suppose we'll update this `catch` block and start rejecting messages with invalid certificates later.

Reply:
Yes it will when OCM validation phase will be over (few days ?).